workflow/setup_expt.py

```diff
@@ -117,90 +103,33 @@ def edit_baseconfig(host, inputs, yaml_dict):
-        "@APP@": inputs.app,
-        "@NMEM_ENS@": getattr(inputs, 'nens', 0)
-    }
-
-    if getattr(inputs, 'nens', 0) > 0:
-        extend_dict['@CASEENS@'] = f'C{inputs.resensatmos}'
+    dict_out = AttrDict({
+        "PSLOT": inputs.pslot,
+        "SDATE": to_YMDH(inputs.idate),
+        "EDATE": to_YMDH(inputs.edate),
+        "CASECTL": f'C{inputs.resdetatmos}',
+        "OCNRES": f"{int(100.*inputs.resdetocean):03d}",
+        "EXPDIR": inputs.expdir,
+        "COMROOT": inputs.comroot,
+        "EXP_WARM_START": is_warm_start,
+        "MODE": inputs.mode,
+        "INTERVAL_GFS": inputs.interval,
+        "SDATE_GFS": to_YMDH(inputs.sdate_gfs),
+        "APP": inputs.app,
+        "NMEM_ENS": getattr(inputs, 'nens', 0),
+        "ICSDIR": inputs.icsdir
+    })
+
+    if dict_out.NMEM_ENS > 0:
+        dict_out.CASEENS = f'C{inputs.resensatmos}'
 
     if inputs.mode in ['cycled']:
-        extend_dict["@DOHYBVAR@"] = "YES" if inputs.nens > 0 else "NO"
-
-    # Further extend/redefine base_dict with extend_dict
-    base_dict = dict(base_dict, **extend_dict)
-
-    # Add/override 'base'-specific declarations in base_dict
-    if 'base' in yaml_dict:
-        base_dict = dict(base_dict, **get_template_dict(yaml_dict['base']))
-
-    base_input = f'{inputs.configdir}/config.base'
-    base_output = f'{inputs.expdir}/{inputs.pslot}/config.base'
-    edit_config(base_input, base_output, host.info, base_dict)
-
-    return
+        dict_out.DOHYBVAR = "YES" if dict_out.NMEM_ENS > 0 else "NO"
 
-
-def edit_config(input_config, output_config, host_info, config_dict):
-    """
-    Given a templated input_config filename, parse it based on config_dict and
-    host_info and write it out to the output_config filename.
-    """
-
-    # Override defaults with machine-specific capabilties
-    # e.g. some machines are not able to run metp jobs
-    host_dict = get_template_dict(host_info)
-    config_dict = dict(config_dict, **host_dict)
-
-    # Read input config
-    with open(input_config, 'rt') as fi:
-        config_str = fi.read()
-
-    # Substitute from config_dict
-    for key, val in config_dict.items():
-        config_str = config_str.replace(key, str(val))
-
-    # Ensure no output_config file exists
-    if os.path.exists(output_config):
-        os.unlink(output_config)
-
-    # Write output config
-    with open(output_config, 'wt') as fo:
-        fo.write(config_str)
-
-    print(f'EDITED:  {output_config} as per user input.')
-
-    return
-
-
-def get_template_dict(input_dict):
-    # Reads a templated input dictionary and updates the output
-
-    output_dict = dict()
-
-    for key, value in input_dict.items():
-        # In some cases, the same config may be templated twice
-        # Prevent adding additional "@"s
-        if "@" in key:
-            output_dict[f'{key}'] = value
-        else:
-            output_dict[f'@{key}@'] = value
-
-    return output_dict
```

Comment:
I think the ctests still make use of this feature.  Should they be updated as well?

Reply:
the scripts under `ctests/` use `cmake` to configure the scripts.  Specifically, `configure_file` is used.  `configure_file` from `cmake` expects the template to be defined as `@VARNAME@`.  It then replaces all instances when `@ONLY` is specified in the call.